README.md

```diff
@@ -49,11 +51,25 @@ Parameter | Description | Required | Default
 `buf_input_https_password` | The password for the repository to check compatibility against. | | [`${{github.token}}`][context]
 `buf_token` | The Buf [authentication token][token] used for private [Inputs][input]. | |
 
-These parameters are derived from [`action.yml`](./action.yml).
+> These parameters are derived from [`action.yml`](./action.yml).
+
+### Constraints
 
-> **Note**: For the `buf-breaking-action` to detect changes successfully, both the `input` and the
-* `against` must be buildable by the `buf` CLI. You can verify this locally using the
-> [`buf build`][buf-build] command on both Inputs.
+For the `buf-breaking-action` to detect changes successfully, both the `input` and the `against`
+need to be properly formed Inputs, that is, `buf` needs to be able to [build][buf-build] both into
+an [Image]. You can verify this locally using the [`buf build`][buf-build] command on both Inputs.
+Some examples:
+
+```sh
+# Build the `main` branch
+buf build .git#main
+
+# Build a feature branch
+buf build .git#v0.1.0
```

Comment:
We can include an example with `branch` and ref`.

```suggestion
buf build .git#branch=main

# Build the v0.1.0 feature tag
buf build .git#ref=v0.1.0
```

Reply:
Feel free to update the comment language - the `branch` and `ref` bits are the important bits.

Reply:
Good call